scipy-1.0/paper.tex

```diff
@@ -185,6 +186,114 @@ \section*{Project organisation and community}
 external representative on the NumFOCUS Board of Directors.
 
 \textbf{Roadmap}
+The SciPy Roadmap\cite{SciPy_roadmap} is a continuously-updated document
+maintained by the community that describes some of the major directions
+for improvement for the project, as well as specific limitations and
+matters that require assistance moving forward.
+
+In general, we encourage changes that improve clarity in the API of the library,
+but strongly discourage breaking backwards compatibility, given our position near
+the base of the scientific Python computing stack.
+
+Unit test coverage for most recent code additions / adjustments is quite
+respectable, with overall project test coverage holding steady at
+around 76\% for the last six months (Figure~\ref{fig:coverage}),
+but some of the historical components of the code base
+would still benefit from increased test coverage. We are still striving
+to increase the number of high-quality SciPy usage tutorials beyond
+our current 15 section offering\cite{SciPy_tutorials}, but our
+standard documentation for library features is already in excellent shape.
+
+\begin{figure}[H]
+\centering
+\includegraphics[width=\textwidth]{static/coverage-chart}
+\caption{\% coverage of the SciPy code base by unit tests over the six
+	 months preceding June 28, 2018 as reported by the \texttt{codecov}
+	 service. The analysis of lines of source code covered is performed
+	 automatically in our continuous integration suite using the
+	 \texttt{pytest-cov} and \texttt{gcov} libraries.}
+\label{fig:coverage}
+\end{figure}
+
+The low-level Cython code in our library (which interacts with C-level
+code and exposes it for Python usage) could use some measure of modernization,
+including migration to typed memoryviews to handle NumPy arrays instead of
+the older syntax:
+
+\begin{lstlisting}[language=Python]
+# example of a function that takes a single
+# 3 dimensional typed memoryview as an argument
+# this allows Cython-level handling of a NumPy
+# array object
+cpdef int func_new(int[:, :, :] arr):
+    pass
+
+# example of the old syntax for handling
+# the same scenario
+cpdef int func_old(object[int, ndim=3, mode='strided'] arr):
+    pass
+
+\end{lstlisting}
+
+\texttt{fftpack} has some single-precision floating point errors, a caching bug,
+and would benefit from the addition of the Bluestein algorithm. There's also
+too much community code duplication with \texttt{numpy.fft}, which
+requires clarification of best choices and long-term plans.
```

Comment:
I propose to remove the first bit, and mention the overlap with numpy for both `fftpack` and `linalg` together